stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -1087,6 +1088,17 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
         _sliceRange = range
     }
     
+    @inline(__always)
+    func _validateIndex(_ index: Int, message: String? = nil) {
```

Comment:
Should 'message' be an autoclosure so that work to generate it isn't evaluated unless it actually fails?